cluster/swarm/cluster.go

```diff
@@ -68,7 +68,7 @@ func NewCluster(scheduler *scheduler.Scheduler, store *state.Store, TLSConfig *t
 	}
 
 	// Set up discovery.
-	cluster.discovery, err = discovery.New(dflag, heartbeat)
+	cluster.discovery, err = discovery.New(dflag, heartbeat, 0)
```

Comment:
How about building the discovery object in manage directly? Dependency injection would be better.

Also, `swarm.discovery.heartbeat`: we could simply reuse the flHeartbeat we already have in manage